### PR TITLE
[FIX] Broken multiple OAuth integrations

### LINKED ISSUE #24704: Can not login into https://open.rocket.chat/ using github
### Description:

I logged into rocket chat using my Github account previously. But after a few months, I changed my device ( operating system ios ) and tried to login into it with GitHub but I was not able to do that. I tried login in from the desktop but was not able to do that also. This issue is with both mobile and desktop (chrome/firefox).

Here is a video of me trying to log in. 

https://user-images.githubusercontent.com/74172466/156782076-37b104ad-d43e-4dbf-aa40-70ba8e9002a2.mp4

### Steps to reproduce:

1. Try login into rocket chat with your GitHub account

### Server Setup Information:

- Version of Rocket.Chat Server: 4.5.0
- Operating System: Windows

### PR DESCRIPTION
**Pull request by debdutdeb**
Signed-off-by: Debdut Chakraborty <debdut.chakraborty@rocket.chat>

## Issue(s)

Closes #24704

## Further comments

- GitHub
- GitLab
- Drupal
- Dolphin
- Tokenpass

### CODE REVIEW

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
maybe this is a good PR to fix them all?

https://github.com/RocketChat/Rocket.Chat/blob/develop/app/dolphin/lib/common.js#L33
https://github.com/RocketChat/Rocket.Chat/blob/develop/app/drupal/lib/common.js#L30
https://github.com/RocketChat/Rocket.Chat/blob/develop/app/tokenpass/lib/common.js#L27

**Comment by debdutdeb** [comment]
Yupie makes sense

File: app/dolphin/lib/common.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ function DolphinOnCreateUser(options, user) {
 
 if (Meteor.isServer) {
 	Meteor.startup(() =>
-		settings.get('Accounts_OAuth_Dolphin_URL', (key, value) => {
+		settings.watch('Accounts_OAuth_Dolphin_URL', (key, value) => {
```
Comment (sampaiodiego) [inline app/dolphin/lib/common.js:33]: `settings.watch` callback receives only the value:
```suggestion
		settings.watch('Accounts_OAuth_Dolphin_URL', (value) => {
```

File: app/drupal/lib/common.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ const Drupal = new CustomOAuth('drupal', config);
 
 if (Meteor.isServer) {
 	Meteor.startup(function () {
-		settings.get('API_Drupal_URL', function (key, value) {
+		settings.watch('API_Drupal_URL', function (key, value) {
```
Comment (sampaiodiego) [inline app/drupal/lib/common.js:30]:
```suggestion
		settings.watch('API_Drupal_URL', function (value) {
```

File: app/github-enterprise/lib/common.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ const GitHubEnterprise = new CustomOAuth('github_enterprise', config);
 
 if (Meteor.isServer) {
 	Meteor.startup(function () {
-		settings.get('API_GitHub_Enterprise_URL', function (key, value) {
+		settings.watch('API_GitHub_Enterprise_URL', function (key, value) {
```
Comment (sampaiodiego) [inline app/github-enterprise/lib/common.js:25]:
```suggestion
		settings.watch('API_GitHub_Enterprise_URL', function (value) {
```

File: app/tokenpass/lib/common.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ const Tokenpass = new CustomOAuth('tokenpass', config);
 
 if (Meteor.isServer) {
 	Meteor.startup(function () {
-		settings.get('API_Tokenpass_URL', function (key, value) {
+		settings.watch('API_Tokenpass_URL', function (key, value) {
```
Comment (sampaiodiego) [inline app/tokenpass/lib/common.js:27]:
```suggestion
		settings.watch('API_Tokenpass_URL', function (value) {
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: app/gitlab/lib/common.js [hunk 5]
```diff
@@ -28,9 +28,9 @@ if (Meteor.isServer) {
 			Gitlab.configure(config);
 		}, 300);
 
-		settings.get('API_Gitlab_URL', updateConfig);
-		settings.get('Accounts_OAuth_Gitlab_identity_path', updateConfig);
-		settings.get('Accounts_OAuth_Gitlab_merge_users', updateConfig);
+		settings.watch('API_Gitlab_URL', updateConfig);
```
Comment (KevLehman) [inline app/gitlab/lib/common.js:31]: Maybe a watchMultiple would be good here? 👀

Comment (sampaiodiego) [reply app/gitlab/lib/common.js:31]: true

**Review by sampaiodiego — APPROVED** [review]

**Comment by u23a** [comment]
It's [said](https://forums.rocket.chat/t/upcoming-changes-to-identity-management-integrations/11994/108) OAuth feature is moved to EE.So I never upgrade to 4.x.x version.
It looks like OAuth feature is added back by the community?

**Pull request merged**